subworkflows/local/enrichment/main.nf

```diff
@@ -7,46 +7,46 @@ include { GPROFILER2_GOST } from "../../../modules/nf-core/gprofiler2/gost/main.
 
 workflow ENRICHMENT {
     take:
-    ch_tools        // [ pathway_name, enrichment_map ]
-    ch_counts
-    ch_results_genewise
-    ch_results_genewise_filtered
-    ch_adjacency
+    ch_counts                       // [ meta, counts] with meta keys: method, args_cor
+    ch_results_genewise             // [ meta, results] with meta keys: method, args_cor
+    ch_results_genewise_filtered    // [ meta, results] with meta keys: method, args_cor
+    ch_adjacency                    // [ meta, adj_matrix] with meta keys: method, args_cor
     // TODO: add ch_gm when provided by user, etc.
 
     main:
 
     // initialize empty results channels
     ch_enriched = Channel.empty()
+    ch_gmt      = Channel.empty()
+
+    ch_adjacency
+        .branch {
+            grea: it[0]["method"] == "grea"
+            gsea: it[0]["method"] == "gsea"
+        }
+        .set { ch_adjacency }
 
     // ----------------------------------------------------
     // Construct gene set database
     // ----------------------------------------------------
 
     // TODO this should be optional, only run when there is no gene set data provided by user
 
+    // empty counts channel of ch_adjacency is empty to skip unnecessary MYGENE computations
```

Comment:
MYGENE is only needed as far as the user does not provide a gmt file, and that a enrichment module is called